tf/utils.go

```diff
@@ -136,6 +138,136 @@ func NewPassReport(plan *tfjson.Plan) types.PassReport {
 	return out
 }
 
+func NewCoverageReportFromState(state *tfjson.State) (types.CoverageReport, error) {
+	out := types.CoverageReport{
+		Coverages: make(map[types.Resource]*coverage.Model, 0),
+	}
+	if state == nil || state.Values == nil || state.Values.RootModule == nil || state.Values.RootModule.Resources == nil {
+		return out, nil
+	}
+	for _, res := range state.Values.RootModule.Resources {
+		if !strings.HasPrefix(res.Address, "azapi_") {
+			continue
+		}
+
+		id := ""
+		if v, ok := res.AttributeValues["id"]; ok {
+			id = v.(string)
+		}
+		resourceType := ""
+		if v, ok := res.AttributeValues["type"]; ok {
+			resourceType = v.(string)
+		}
+
+		body := map[string]interface{}{}
+		if bodyRaw, ok := res.AttributeValues["body"].(string); ok {
+			if value, ok := res.AttributeValues["tags"]; ok && value != nil && len(value.(map[string]interface{})) > 0 {
+				tagsModel := value.(map[string]interface{})
+				if len(tagsModel) != 0 {
```

Comment:
>len(tagsModel) != 0 [](http://example.com/codeflow?start=7&length=19)

already judged `len(value.(map[string]interface{})) > 0` in outer `if`

Reply:
Thanks, I have changed the code.